src/nft/queries/genie/RouteFetcher.ts

```diff
@@ -53,13 +53,13 @@ const buildRouteItem = (item: GenieAsset): RouteItem => {
     id: item.id,
     symbol: item.priceInfo.baseAsset,
     name: item.name,
-    decimals: item.decimals || 0, // 0 for fungible items
+    decimals: parseFloat(item.priceInfo.baseDecimals),
     address: item.address,
     tokenType: item.tokenType,
     tokenId: item.tokenId,
     marketplace: item.marketplace,
     collectionName: item.collectionName,
-    amount: item.amount || 1, // default 1 for a single asset
+    amount: 1,
```

Comment:
is this the same limit 1 in the graphql query?

Reply:
No this is a leftover from when the contract supported swapping pooled NFTs